### PR TITLE
Bring in changes from another project instance

### PR DESCRIPTION
**Pull request by derekderaps**

### CODE REVIEW

File: .environment [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Statements in this file will be executed (sourced) by the shell in SSH
+# sessions, in deploy hooks, in cron jobs, and in the application's runtime
+# environment.
+
+# Allow executable app dependencies from Composer to be run from the path.
+export PATH=/app/vendor/bin:$PATH
```
Comment (RobLoach) [inline .environment]: `/app/vendor/bin`? Anything in this file seems like it should just reference the project path. All systems, for instance, won't have `/app`. Could you add this to your `~/.bashrc`, `.zshrc` or similar?

Another solution around this could be to use [Composer Scripts](https://getcomposer.org/doc/articles/scripts.md), since they [automatically add vendor/bin to PATH](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands).

```
{
    "scripts": {
        "test": "phpunit"
    }
}
```
```
composer test
```

File: package.json [hunk 2]
```diff
@@ -23,6 +24,13 @@
...
     "grunt": "^1.0.1",
     "grunt-build-control": "^0.7.1",
-    "grunt-cli": "^1.2.0"
+    "grunt-cli": "^1.2.0",
+    "kalastatic": "^3.1.0"
+  },
+  "dependencies": {
+    "bootstrap-sass": "^3.3.7",
+    "font-awesome": "^4.7.0",
+    "kss-bootstrap": "^4.0.0",
```
Comment (RobLoach) [inline package.json]: Will every project use Bootstrap? I suppose so!

File: composer.json [hunk 3]
```diff
@@ -65,10 +68,21 @@
         ],
         "post-install-cmd": [
             "DrupalProject\\composer\\ScriptHandler::createRequiredFiles",
-            "@drupal-scaffold"
+            "@drupal-scaffold",
+            "@kalastatic"
```
Comment (RobLoach) [inline composer.json]: KalaStatic may not be installed on every project, but this does seem useful. Let's keep 'er in! :+1:

File: phpunit.xml.dist [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+-->
+  <php>
+    <!-- Set error reporting to E_ALL. -->
+    <ini name="error_reporting" value="32767"/>
+    <!-- Do not limit the amount of memory tests take to run. -->
+    <ini name="memory_limit" value="-1"/>
+    <!-- Example SIMPLETEST_BASE_URL value: http://localhost -->
+    <env name="SIMPLETEST_BASE_URL" value="http://nginx:80"/>
+    <!-- Example SIMPLETEST_DB value: mysql://username:password@localhost/databasename#table_prefix -->
+    <env name="SIMPLETEST_DB" value="mysql://drupal:drupal@mariadb/"/>
```
Comment (RobLoach) [inline phpunit.xml.dist]: Note that this will be different in DrupalVM. This right now targets Docker's mariadb creds.

File: composer.json [hunk 5]
```diff
@@ -65,10 +68,21 @@
...
         ],
         "post-update-cmd": [
             "DrupalProject\\composer\\ScriptHandler::createRequiredFiles"
+        ],
+        "kalastatic": [
+            "npm install",
+            "npm test"
```
Comment (RobLoach) [inline composer.json]: Stuck `npm test` in the `composer install` script on FTUSA because their robots weren't smart enough to run `npm test` for us. Recommend we move `npm test` after `composer test`, right after the "CALL PHPUNIT TESTS HERE".

File: composer.json (hunk 3 above)
Comment (RobLoach) [inline composer.json]: On a second thought, may be good to use `npm install` instead?  Make it just use `npm` to manage whatevs in there. Is there a way to say....

"Run `npm install` if `npm` is available."?

File: .environment (hunk 1 above)
Comment (derekderaps) [inline .environment]: Yeah, sorry. Jay added that one to the FTUSA repo and I blindly carried it over... Deleting it nowsies!

File: package.json (hunk 2 above)
Comment (derekderaps) [inline package.json]: Yeah I was pretty liberal with how much opinionated build settings were included. We can definitely take some out though, if you think I went too far. I figure if more than half of our sites will use it, then it makes sense. The rest can take it out.

File: phpunit.xml.dist (hunk 4 above)
Comment (derekderaps) [inline phpunit.xml.dist]: okay, hmmm. Is there a good way to manage this so that it can be specified locally / per environment?

File: composer.json (hunk 5 above)
Comment (derekderaps) [inline composer.json]: okay, cool, will do!

File: phpunit.xml.dist (hunk 4 above)
Comment (RobLoach) [inline phpunit.xml.dist]: Unsure if there's a way to make SIMPLETEST_DB "just work" across any environment. Is there a way to define environment variables in the DrupalVM configs? We could move this definition over there?

Comment (derekderaps) [inline phpunit.xml.dist]: Yeah I don't think "just work" is an option. Let me rephrase, do you know if there's any way for the PHPUnit config to use variables of some sort? Environment variables, perhaps?

**Comment by derekderaps** [comment]
Let me know if you see any other changes needed, @RobLoach. Thanks!

File: phpunit.xml.dist (hunk 4 above)
Comment (RobLoach) [inline phpunit.xml.dist]: For now, I've moved this to `phpunit.xml.dist.sample`.

File: composer.json [hunk 6]
```diff
@@ -65,10 +68,18 @@
         ],
         "post-install-cmd": [
             "DrupalProject\\composer\\ScriptHandler::createRequiredFiles",
-            "@drupal-scaffold"
+            "@drupal-scaffold",
+            "if hash npm; then npm install; fi"
```
Comment (RobLoach) [inline composer.json]: Oh fancy!

File: composer.json [hunk 7]
```diff
@@ -65,10 +68,18 @@
...
         ],
         "post-update-cmd": [
             "DrupalProject\\composer\\ScriptHandler::createRequiredFiles"
+        ],
+        "test": [
+            "phpcs --config-set installed_paths vendor/drupal/coder/coder_sniffer",
+            "phpcs --standard=Drupal web/modules/custom",
+            "phpcs --standard=Drupal web/themes",
+            "echo 'CALL PHPUNIT TESTS HERE'",
+            "if hash npm; then npm test; fi"
```
Comment (RobLoach) [inline composer.json]: This hash usage is #awesome.

File: web/sites/dev [hunk 8]
```diff
@@ -0,0 +1 @@
+default/
```
Comment (RobLoach) [inline web/sites/dev]: What is this file?

Comment (derekderaps) [inline web/sites/dev]: Good question, it's a symlink to the `default` sites folder. Since we have `live` and `test` folders, I thought we should have `dev`, too.

Comment (RobLoach) [inline web/sites/dev]: SMRT.
